pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -134,13 +166,65 @@ func createService(componentMeta metav1.ObjectMeta, componentExt *v1beta1.Compon
 	return service
 }
 
+func createHeadSvc(componentMeta metav1.ObjectMeta) *corev1.Service {
+	workerComponentMeta := componentMeta.DeepCopy()
+	predictorSvcName := workerComponentMeta.Name
+	isvcGeneration := componentMeta.GetLabels()[constants.InferenceServiceGenerationPodLabelKey]
+	workerComponentMeta.Name = constants.GeHeadServiceName(predictorSvcName, isvcGeneration)
+	workerComponentMeta.Labels[constants.MultiNodeRoleLabelKey] = constants.MultiNodeHead
+
+	service := &corev1.Service{
+		ObjectMeta: *workerComponentMeta,
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": constants.GetRawServiceLabel(predictorSvcName),
+				constants.InferenceServiceGenerationPodLabelKey: isvcGeneration,
+			},
+			ClusterIP:                "None", // Without this, it requires a Port but this Service does not need it.
+			PublishNotReadyAddresses: true,
+		},
+	}
+	return service
+}
+
+func (r *ServiceReconciler) cleanHeadSvc() error {
+	svcList := &corev1.ServiceList{}
+	if err := r.client.List(context.TODO(), svcList, client.MatchingLabels{
+		constants.MultiNodeRoleLabelKey: constants.MultiNodeHead,
+	}); err != nil {
+		return err
+	}
+
+	sort.Slice(svcList.Items, func(i, j int) bool {
+		return svcList.Items[i].CreationTimestamp.Time.After(svcList.Items[j].CreationTimestamp.Time)
+	})
+
+	// Keep the 3 newest services and delete the rest
+	for i := 3; i < len(svcList.Items); i++ {
```

Comment:
why are there 3 svc head services?

Reply:
I am confused too, head svc still belongs to a isvc so it can be GCed once it is deleted

Reply:
This is for upgrade usecase. For example, each worker node should point to unique head node. For that, it should create a head svc per each worker node. 

example) 
first worker node pod  ----------->  first head node svc ----------------------> first head node pod

second worker node pod -------------> second head node svc -------------> second head node pod

This is for no shutdown time using deploymentStragtegy(rollingUpdate)

Then if there are more than 3 head svcs, it will remove old ones.